src/Microsoft.AspNetCore.JsonPatch/Internal/DynamicObjectAdapter.cs

```diff
@@ -146,16 +130,19 @@ public class DynamicObjectAdapter : IAdapter
             object target,
             IContractResolver contractResolver,
             string segment,
-            out object property)
+            out object property,
+            out string errorMessage)
         {
             var jsonDynamicContract = contractResolver.ResolveContract(target.GetType()) as JsonDynamicContract;
 
             var propertyName = jsonDynamicContract?.PropertyNameResolver(segment);
 
-            var binder = CSharp.RuntimeBinder.Binder.GetMember(CSharpBinderFlags.None,
-                propertyName,
+            var binder = CSharp.RuntimeBinder.Binder.GetMember(
+                CSharpBinderFlags.None,
+                propertyName ?? segment,
```

Comment:
@rynowak  If `segment` is null it will throw [here](https://github.com/aspnet/JsonPatch/blob/dev/src/Microsoft.AspNetCore.JsonPatch/Adapters/ObjectAdapter.cs#L129) before it gets to the adapter.

If the `contract` or the `propertName` cannot be resolved, segment may still be a valid path for the patch document. Is that risky?